src/utils/daos_hdlr.c

```diff
@@ -1645,6 +1646,80 @@ fs_copy_file(struct cmd_args_s *ap,
 	return rc;
 }
 
+#define SUB_STR_LEN 256
+static int
+fs_copy_symlink(struct cmd_args_s *ap,
+		struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+		struct stat *src_stat,
+		const char *src_path,
+		const char *dst_path)
+{
+	int		rc;
+	daos_size_t	len = 0;
+	char		symlink_value[PATH_MAX-SUB_STR_LEN];
+
+	len = PATH_MAX - SUB_STR_LEN - 1;
+	if (src_file_dfs->type == POSIX) {
+		len = readlink(src_path, symlink_value, (size_t)len);
+		if (len < 0) {
+			rc = daos_errno2der(errno);
+			DH_PERROR_SYS(ap, rc, "fs_copy_symlink failed on readlink('%s')",
+				      src_path);
+			D_GOTO(out_copy_symlink, rc);
+		}
+	} else if (src_file_dfs->type == DAOS) {
+		rc = dfs_sys_readlink(src_file_dfs->dfs_sys, src_path, symlink_value, &len);
+		if (rc != 0) {
+			rc = daos_errno2der(rc);
+			DH_PERROR_SYS(ap, rc, "fs_copy_symlink failed on dfs_sys_readlink('%s')",
+				      src_path);
+			D_GOTO(out_copy_symlink, rc);
+		}
+		/*length of symlink_value includes the NULL terminator already.*/
+		len--;
+	} else {
+		rc = daos_errno2der(EINVAL);
+		DH_PERROR_DER(ap, rc, "unknown type for %s", src_path);
+		D_GOTO(out_copy_symlink, rc);
+	}
+	symlink_value[len] = 0;
+
+	if (dst_file_dfs->type == POSIX) {
+		rc = symlink(symlink_value, dst_path);
+		if ((rc != 0) && (errno == EEXIST)) {
+			D_WARN("Link %s exists.\n", dst_path);
```

Comment:
D_WARN isn't used elsewhere in this file for anything, why not DH_PERROR_SYS like other cases?

Reply:
Both DH_PERROR_DER and DH_PERROR_SYS generate error messages like, 

ERROR: : Link //hello_s_s_s exists.
ERROR: : : File exists (17)

I would prefer printing warning only.